src/core/src/op/space_to_batch.cpp

```diff
@@ -55,8 +55,8 @@ void SpaceToBatch::validate_and_infer_types() {
                           "pads_end must be an integral number but got (",
                           pads_end_type,
                           ").");
-    const auto output_shape = shape_infer(this, ov::util::get_node_input_partial_shapes(*this)).front();
-    set_output_type(0, data_type, output_shape);
+    const auto& output_shapes = shape_infer(this, ov::util::get_node_input_partial_shapes(*this));
```

Comment:
In so many places with `shape_infer`, there are `auto& output_shapes` or `auto output_shapes` used. Was it intentional? May it be aligned?

Reply:
Yes will fix that. For store vector it should be fine to have `const auto output_shapes`.